src/communication/communication_channel.cpp

```diff
@@ -174,6 +173,7 @@ namespace cubcomm
 	return REQUEST_REFUSED;
       }
 
+    m_type = CHANNEL_TYPE::INITIATOR;
```

Comment:
Why is it moved?

Reply:
it seems weird that only `m_type` initialized even if m_socket is failed to open. 

Not related this issue, but just a small refactoring.